### DISCUSSION TITLE
Which GOG version of the game should I use?

### DISCUSSION THREAD

**Discussion by hvcrios** [ihhub/fheroes2#5669] [Q&A] [answered]
edit: I'm running on linux using Flatpak btw, should have mentioned it earlier.

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------

edit2: For the folks running it on linux, using the flatpak, easiest way is to grab the zipped release for linux, there's a "extract_homm2_resources.sh" script in there, if you run it without supplying a destination folder (would be argument 2,  1 being the HoMM2 install folder ), it will drop the files in a "./local/share/fheroes2" folder, then you can just move that over to your ".var/app/io.github.ihhub.Fheroes2/data/" folder.

Or of course just run the script with the arguments to begin it and have it drop the files directly there. so the command would be like:

`./extract_homm2_resources.sh "HoMM2 install folder" "/home/user/.var/app/io.github.ihhub.Fheroes2/data/fheroes2"`

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------

op:

GOG has 2 versions of it, the standard one it installs with galaxy using DosBox and one which is a "windows version" you can download separately, they both have different data files.

More importantly the dosbox version does not have the "ANIM" subfolder at all, whilst the windows one does, but the windows one is missing a bunch of the music files, and some files in the "DATA" folder, that are in the dosbox version.

So which one should I use the files from?

**Comment by zenseii** [comment, marked as answer]
Hi, @hvcrios.

Both of those versions work equally fine with fheroes2. The reason they might seem to have different files is because the DOSBox version has a disc image file called heroes2.gog (which you can rename to heroes2.bin and mount). The installer works with both and will grab all necessary files provided your Heroes 2 installation is normal. There's also the manual way to copy the folders you need.

I believe the windows version has all the cd music files but if not then you can download these separately from gog in the higher quality flac format. There is a script you can simply that will help you import these to fheroes2.

Note that you should install fheroes2 in its own separate directory than your Heroes 2 installation.

**Reply by hvcrios** [reply]
edit: NVM! I read the install instructions(imagine that!) and saw that the script is supplied with the regular zip distribution.

I think I'll ditch the flatpak one for that, or just grab the script and see which files it gets, thx for the answer :)

--------------------------------------------------------------------------------------------------------------------------------------------------------------

I see, the thing is I didn't use an installer, I'm running it on Linux, and using the Flatpak version.

All I got was a prompt when trying to launch telling me which folders were needed, and so I just copied them over from the galaxy installation (which I run through lutris/wine), then I was a bit confused since I couldn't find the "ANIM" folder, downloaded and installed the "windows version" and saw the folder was there.

Basically i'm using all the other data from the Dos version, and only the ANIM folder from the windows one :P 

Seems to be working fine, is there a better way to do it? you mentioned a script I could use to import the necessary files for the kickass music, how do I use that?

**Reply by zenseii** [reply]
I see. I assumed you were on Windows since it is a powershell script so only for Windows. Maybe you can run it through lutris/wine? It is called `resource_extraction_toolset.bat` btw.

I don't remember exactly how this script works but drag 'n' drop should be supported, so I guess you can just drop the compressed flac archive over the script file and it will give you a `music.flac` folder that you should just rename to `music`.

If this doesn't work you can also grab the `music` folder from the dosbox installation.

**Reply by hvcrios** [reply]
Ah yeah, the kind folks provide a shell script for the linux version as well, for grabbing the resources I mean, not the flac conversion, so I just grabbed that from the zipped release, and ran it, just had to install a dependency (bsdtar) and it worked, I think what I had done had already solved it (grabbed the anim folder from the windows version), but this is more neat.

I'll take a look at that powershell script and see if i can replicate it in linux.

I think this has all the music in it though.

Thx for the help :)

**Reply by zenseii** [reply]
No worries.

The script I'm talking about is specifically if you want to use the higher quality flac files provided with GoG or if you have the music in mp3 format for some reason.
